tests/src/main/java/io/apicurio/tests/utils/subUtils/TestUtils.java

```diff
@@ -0,0 +1,108 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestUtils.class);
+
+    private TestUtils() {
+        // All static methods
+    }
+
+    /**
+     * Poll the given {@code ready} function every {@code pollIntervalMs} milliseconds until it returns true,
+     * or throw a TimeoutException if it doesn't returns true within {@code timeoutMs} milliseconds.
+     * @return The remaining time left until timeout occurs
+     * (helpful if you have several calls which need to share a common timeout),
+     * */
+    public static long waitFor(String description, long pollIntervalMs, long timeoutMs, BooleanSupplier ready) throws TimeoutException {
+        return waitFor(description, pollIntervalMs, timeoutMs, ready, () -> { });
+    }
+
+    public static long waitFor(String description, long pollIntervalMs, long timeoutMs, BooleanSupplier ready, Runnable onTimeout) throws TimeoutException {
+        LOGGER.debug("Waiting for {}", description);
+        long deadline = System.currentTimeMillis() + timeoutMs;
+        while (true) {
+            boolean result;
+            try {
+                result = ready.getAsBoolean();
+            } catch (Exception e) {
+                result = false;
+            }
+            long timeLeft = deadline - System.currentTimeMillis();
+            if (result) {
+                return timeLeft;
+            }
+            if (timeLeft <= 0) {
+                onTimeout.run();
+                TimeoutException exception = new TimeoutException("Timeout after " + timeoutMs + " ms waiting for " + description);
+                exception.printStackTrace();
+                throw exception;
+            }
+            long sleepTime = Math.min(pollIntervalMs, timeLeft);
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("{} not ready, will try again in {} ms ({}ms till timeout)", description, sleepTime, timeLeft);
+            }
+            try {
+                Thread.sleep(sleepTime);
+            } catch (InterruptedException e) {
+                return deadline - System.currentTimeMillis();
+            }
+        }
+    }
+
+    /**
+     * Method to create and write String content file
+     * @param filePath path to file
+     * @param text content
+     */
+    public static void writeFile(String filePath, String text) {
+        Writer writer = null;
+        try {
+            writer = new BufferedWriter(new OutputStreamWriter(
+                    new FileOutputStream(filePath), StandardCharsets.UTF_8));
+            writer.write(text);
+        } catch (IOException e) {
+            LOGGER.info("Exception during writing text in file");
+            e.printStackTrace();
+        } finally {
+            try {
+                if (writer != null) {
+                    writer.close();
+                }
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+    }
+
+    public static String ruleToString(Rule rule) {
```

Comment:
I'm sure there is some JSON lib on the classpath, so you don't need to do this manually.